### PR TITLE
NGINX data connector

### PR DESCRIPTION
**Pull request by vu-socprime**
Added NGINX data connector

### CODE REVIEW

File: DataConnectors/NGINX/Connector_NGINX_agent.json [hunk 1]
```diff
@@ -0,0 +1,149 @@
+{
+    "id": "NGINX HTTP Server",
```
Comment (v-admahe) [inline DataConnectors/NGINX/Connector_NGINX_agent.json:2]: Id should be in this format NGINXHTTPServer (ProviderNameApplianceName).Please refer sample template https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/Templates/Connector_REST_API_template.json

File: DataConnectors/NGINX/Connector_NGINX_agent.json [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    "title": "NGINX HTTP Server",
+    "publisher": "Nginx",
+    "descriptionMarkdown": "NGINX HTTP Server data connector provides the capability to ingest [NGINX HTTP Server](https://nginx.org/en/) events into Azure Sentinel. Refer to [NGINX Logs documentation](https://nginx.org/en/docs/http/ngx_http_log_module.html) for more information.",
+    "additionalRequirementBanner": "This data connector depends on a parser based on Kusto Function to work as expected. Follow the steps to use this Kusto Function alias **NGINXHTTPServer** in queries and workbooks. [Follow steps to get this Kusto Function>](https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/NGINX/NGINXHTTPServer.txt)",
+    "graphQueries": [
+        {
+            "metricName": "Total data received",
+            "legend": "NGINX Events",
```
Comment (v-admahe) [inline DataConnectors/NGINX/Connector_NGINX_agent.json:10]: Legend should be datatype name that is NGINX_CL. Please refer sample template https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/Templates/Connector_REST_API_template.json

File: DataConnectors/NGINX/Connector_NGINX_agent.json [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    "connectivityCriterias": [
+        {
+            "type": "IsConnectedQuery",
+            "value": [
+                "NGINXHTTPServer\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
+            ]
+        }
+    ],
+    "availability": {
+        "status": 1
```
Comment (v-admahe) [inline DataConnectors/NGINX/Connector_NGINX_agent.json:35]: isPreview tag is missing please refer sample template https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/Templates/Connector_REST_API_template.json

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: DataConnectors/NGINX/Connector_NGINX_agent.json [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+            {
+                "provider": "Microsoft.OperationalInsights/workspaces",
+                "permissionsDisplayText": "write permission is required.",
+                "providerDisplayName": "Workspace",
+                "scope": "Workspace",
+                "requiredPermissions": {
+                    "write": true,
+                    "delete": true
+                }
+            }
```
Comment (v-admahe) [inline DataConnectors/NGINX/Connector_NGINX_agent.json:48]: Permission tag is not complete Microsoft.OperationalInsights/workspaces/sharedKeys is missing please refer the sample template
"permissions": {
        "resourceProvider": [
            {
                "provider": "Microsoft.OperationalInsights/workspaces",
                "permissionsDisplayText": "read and write permissions are required.",
                "providerDisplayName": "Workspace",
                "scope": "Workspace",
                "requiredPermissions": {
                    "write": true,
                    "read": true,
                    "delete": true
                }
            },
            {
                "provider": "Microsoft.OperationalInsights/workspaces/sharedKeys",
                "permissionsDisplayText": "read permissions to shared keys for the workspace are required. [See the documentation to learn more about workspace keys](https://docs.microsoft.com/azure/azure-monitor/platform/agent-windows#obtain-workspace-id-and-key)",
                "providerDisplayName": "Keys",
                "scope": "Workspace",
                "requiredPermissions": {
                    "action": true
                }
            }
        ]"
 https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/Templates/Connector_REST_API_template.json

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: DataConnectors/NGINX/Connector_NGINX_agent.json [hunk 5]
```diff
@@ -0,0 +1,149 @@
+{
+    "id": "NGINX HTTP Server",
+    "title": "NGINX HTTP Server",
+    "publisher": "Nginx",
+    "descriptionMarkdown": "NGINX HTTP Server data connector provides the capability to ingest [NGINX HTTP Server](https://nginx.org/en/) events into Azure Sentinel. Refer to [NGINX Logs documentation](https://nginx.org/en/docs/http/ngx_http_log_module.html) for more information.",
+    "additionalRequirementBanner": "This data connector depends on a parser based on Kusto Function to work as expected. Follow the steps to use this Kusto Function alias **NGINXHTTPServer** in queries and workbooks. [Follow steps to get this Kusto Function>](https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/NGINX/NGINXHTTPServer.txt)",
```
Comment (v-admahe) [inline DataConnectors/NGINX/Connector_NGINX_agent.json:6]: Please replace parser link with this link : https://aka.ms/sentinel-NGINXHTTP-parser

File: DataConnectors/NGINX/Connector_NGINX_agent.json [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+                    "write": true,
+                    "delete": true
+                }
+            }
+        ]
+    },
+    "instructionSteps": [
+        {
+            "title": "",
+            "description": ">**NOTE:** This data connector depends on a parser based on a Kusto Function to work as expected. [Follow these steps](https://github.com/Azure/Azure-Sentinel/blob/master/Parsers/NGINX/NGINXHTTPServer.txt) to create the Kusto Functions alias, **NGINXHTTPServer**",
```
Comment (v-admahe) [inline DataConnectors/NGINX/Connector_NGINX_agent.json:54]: Please replace parser link with this link : https://aka.ms/sentinel-NGINXHTTP-parser

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: DataConnectors/NGINX/Connector_NGINX_agent.json (hunk 1 above)
Comment (vu-socprime) [reply DataConnectors/NGINX/Connector_NGINX_agent.json]: updated

File: DataConnectors/NGINX/Connector_NGINX_agent.json (hunk 2 above)
Comment (vu-socprime) [reply DataConnectors/NGINX/Connector_NGINX_agent.json]: updated

File: DataConnectors/NGINX/Connector_NGINX_agent.json (hunk 3 above)
Comment (vu-socprime) [reply DataConnectors/NGINX/Connector_NGINX_agent.json]: updated

File: DataConnectors/NGINX/Connector_NGINX_agent.json (hunk 4 above)
Comment (vu-socprime) [reply DataConnectors/NGINX/Connector_NGINX_agent.json:58]: updated

File: DataConnectors/NGINX/Connector_NGINX_agent.json (hunk 5 above)
Comment (vu-socprime) [reply DataConnectors/NGINX/Connector_NGINX_agent.json]: updated

File: DataConnectors/NGINX/Connector_NGINX_agent.json (hunk 6 above)
Comment (vu-socprime) [reply DataConnectors/NGINX/Connector_NGINX_agent.json]: updated

File: DataConnectors/NGINX/Connector_NGINX_agent.json (hunk 4 above)
Comment (v-admahe) [reply DataConnectors/NGINX/Connector_NGINX_agent.json:58]: Read permission is still missing and permissiondisplaytext should exactly match with the text given in the sample json file.

Please refer the sample template: https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/DataConnectors/Templates/Connector_REST_API_template.json

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Review by v-admahe — APPROVED** [review]

File: DataConnectors/NGINX/Connector_NGINX_agent.json [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+                "scope": "Workspace",
+                "requiredPermissions": {
+                    "read": true,
+                    "write": true,
+                    "delete": true
+                }
+            },
+            {
+                "provider": "Microsoft.OperationalInsights/workspaces/sharedKeys",
+                "permissionsDisplayText": "read permissions to shared keys for the workspace are required. [See the documentation to learn more about workspace keys](https://docs.microsoft.com/azure/azure-monitor/platform/agent-windows#obtain-workspace-id-and-key)",
```
[the remark this answers is not in the document]
Comment (v-admahe) [reply DataConnectors/NGINX/Connector_NGINX_agent.json:53]: @vu-socprime 

Test cases are failing due to dot is missing at the end of permission display text so could you please add full stop at the end of the sentence.

Use this line: read permissions to shared keys for the workspace are required. See the documentation to learn more about workspace keys.

Thanks,
Aditi Maheshwari

[the remark this answers is not in the document]
Comment (v-rucdu) [reply DataConnectors/NGINX/Connector_NGINX_agent.json:53]: @vu-socprime : Could you please resolve review comment.... Thanks!!!

**Review by v-admahe — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**